pkg/repository/inmemory/repository.go

```diff
@@ -3,42 +3,58 @@ package inmemory
 import (
 	"context"
 	"errors"
+	"reflect"
 	"sync"
 
 	"github.com/ThreeDotsLabs/esja/pkg/aggregate"
 	"github.com/ThreeDotsLabs/esja/pkg/repository"
 )
 
-type Repository[T any] struct {
+type Repository[T aggregate.Aggregate[T]] struct {
```

Comment:
does it have to be `T aggregate.Aggregate[T]` or could it be `T aggregate.Aggregate[S]`? 🤔

Reply:
or is it just a question of syntax

Reply:
an interesting question – which parts of repository would then depend on `T` and which on `S` 

I mean, we can stay with `T/T`, but:
1) Does it have to be `T Aggregate[T]` because of logical reasons?
2) If not, would `T Aggregate[S]` be more readable/easier to understand?

Reply:
It's kind of weird, but both `T` and `Aggregate[T]` are the same type. We want T to implement Aggregate, but since it's generic, we need to pass the type. To be honest I was surprised this even worked but it did.

We could do S, but then the interface would be weird, like:

```go
NewRepository[*Postcard, *Postcard]()
```

Reply:
Yeah, the docs are something that still needs polishing. But I'm not sure if the API is stable at this point, we're still experimenting.